src/mongocxx/pool.cpp

```diff
@@ -39,20 +39,20 @@ void pool::_release(client* client) {
 
 pool::~pool() = default;
 
-pool::pool(const uri& uri, stdx::optional<options::ssl> ssl_options)
+pool::pool(const uri& uri, const options::client& options)
```

Comment:
I'm sympathetic to that argument. However, keep ABI in mind. If we use options::client here, and later need to move to options::pool, that would be an ABI break. However, if we start with options::pool, and later need to add behavior to options::pool, that at least has the *possibility* to not be an ABI break, depending on how we implement it.